### DISCUSSION TITLE
Colour coding

### DISCUSSION THREAD

**Discussion by MrOccam** [jonasthewolf/gsn2x#261] [Ideas]
Usually, I would add colour coding when producing GSNs manually, to reporting which goals have been met. I'm attempting to colour code elements of the GSNs I am producing using gsn2x. How could colour coding of elements be implemented using style sheets?

**Comment by jonasthewolf** [comment]
Hi @MrOccam 

my idea would be to add extra classes, and colorise them via a CSS stylesheet.

Please see the [documentation](https://jonasthewolf.github.io/gsn2x/advanced_usecases.html) for how to do this.

Best regards,

Jonas

**Comment by jonasthewolf** [comment]
I created #262 to create such an example.

**Comment by jonasthewolf** [comment]
@MrOccam Thanks for the feedback.

**Comment by jonasthewolf** [comment]
@MrOccam would you mind having a look at the new version [v2.7.3](https://github.com/jonasthewolf/gsn2x/releases/tag/v2.7.3) and its documentation. You might be interested in [the info on stylesheets](https://jonasthewolf.github.io/gsn2x/adv_stylesheets.html).
Your feedback is very much appreciated.

**Comment by MrOccam** [comment]
Hi Jonas,
I can add colour to Goal and Strategy. But cannot add colour to Assumption, Justification, Solution, Context.
It would be helpful to add colour to Solutions. Is this something I'm doing wrong or have you not added support for this yet?

**Comment by jonasthewolf** [comment]
Thanks for the feedback. It is actually possible to also style all other elements.
I updated the [documentation](https://jonasthewolf.github.io/gsn2x/adv_stylesheets.html) to be readily used for that.
I also found a small bug that prevented the documented CSS classes from being added to the corresponding elements.
It is fixed in [v2.7.4](https://github.com/jonasthewolf/gsn2x/releases/tag/v2.7.4).

**Reply by MrOccam** [reply]
After testing v2.7.4 there are still issues with Solutions, text colouring can be changed but background colouring still cannot be changed. 

The yaml file is calls the Solution as:
`Sn1:`
 `text: Hazard Identication Working Group Minutes`
 `classes: [status-red]`

The CSS is set up as follows:
`.status-green path { fill: green; }`
`.status-green text,a { fill: black; }`
`.status-amber path { fill: orange; }`
`.status-amber text,a { fill: black; }`
`.status-red path { fill: red; }`
`.status-red text,a { fill: white; }`

**Reply by jonasthewolf** [reply]
v2.7.4 only fixed a bug that prevented adding custom classes.
v2.8.0 unified the way that you can style the elements.
So, your example works with v2.8.0.

**Reply by MrOccam** [reply]
After testing v2.8.1 with the same code the problem seems to have flipped, now the background can be changed but text cannot

**Reply by jonasthewolf** [reply]
Thanks for the feedback. Unfortunately, I cannot reproduce your issue here.

Let me show you my minimal example:

```yaml
G1:
  text: A
  supportedBy: [Sn1]

Sn1:
  text: Hazard Identication Working Group Minutes
  classes: [status-red]
```

Stylesheet:

```css
.status-green path { fill: green; }
.status-green text,a { fill: black; }
.status-amber path { fill: orange; }
.status-amber text,a { fill: black; }
.status-red path { fill: red; }
.status-red text,a { fill: white; }
```

Excerpt of the resulting SVG:

```xml
<g class="gsnelem gsnsltn status-red gsn_module_localtests_minsn_yaml" id="node_Sn1">
<title>
Sn1
</title>
<path class="border" d="M20,175 a69,69,0,1,0,138,0 a69,69,0,1,0,-138,0 z" fill="none" stroke="black" stroke-width="1"/>
<text font-family="Arial" font-size="12" font-weight="bold" x="56" y="138">
Sn1
</text>
<text font-family="Arial" font-size="12" x="56" y="155">
Hazard
</text>
<text font-family="Arial" font-size="12" x="56" y="167">
Identication
</text>
<text font-family="Arial" font-size="12" x="56" y="179">
Working
</text>
<text font-family="Arial" font-size="12" x="56" y="191">
Group
</text>
<text font-family="Arial" font-size="12" x="56" y="203">
Minutes
</text>
</g>
```

It looks the following:

<img width="477" alt="image" src="https://github.com/jonasthewolf/gsn2x/assets/16023405/326a0b5f-3e3f-4e0d-9724-e3720c546b87">

I today encountered a bug that incorrectly imported a CSS stylesheet (brackets were added that should not be there).
However, if you embedded the CSS, this should not have any impact.
The bug is fixed in 2.8.2 (which I just released).

Please let me know if that helped or you still encounter problems with styling.

**Reply by MrOccam** [reply]
Thanks Jonas, I had more luck with it today. It's working with v2.8.1

**Reply by jonasthewolf** [reply]
That's great to hear! Thanks!

**Comment by jonasthewolf** [comment]
With #271 I made styling elements easier, because all elements are now represented as a `<path>` SVG element.

**Comment by Rlaks10** [comment]
Is it possible to support restructedtext or markdown encoding in the text field? 

something like this,

`G1:`
`     text: *Data* should have _integrity_. ´Link<www.example.com>´_. `
`    encodingType: rst`

This would be very helpful to link the claims or evidences shown externally.

**Reply by jonasthewolf** [reply]
I actually have a long standing feature open to support some markdown formatting (see #27). Unfortunately, it is not that easy to implement without additional dependencies.

However, I was not planning to support links in particular.
They are already supported 😉 just use the `url:` attribute.
This link is clickable from the SVG then, and it will appear in the generated list of evidences.

Does that help already?

**Reply by Rlaks10** [reply]
Thanks and yes having url: already helps. I missed that in documentation.
